type-check.ts

```diff
@@ -246,6 +405,12 @@ function tcDestructure(
     let { target, starred, ignore } = aTarget;
     const tTarget = tcAssignable(env, locals, target);
     const targetType = tTarget.a;
+
+    if (expr.tag === "lambda") {
+      tcLambda(locals, expr, targetType);
+      valueType = tcExpr(env, locals, expr).a;
+    }
+
```

Comment:
Can you say why this actually needed to be hard coded into `tcDestructure` rather than passing in the value type itself??